### PR TITLE
onboard ecto_sql sql.exs integration test, various fixes

### PR DESCRIPTION
**Pull request by kevinlang**
Onboarded 17 new tests! :)

### CODE REVIEW

File: lib/exqlite/connection.ex [hunk 1]
```diff
@@ -439,6 +439,12 @@ defmodule Exqlite.Connection do
 
   defp maybe_changes(_, _), do: nil
 
+  # TODO: should we change the NIF to just return the string
+  defp normalize_columns(columns) do
```
Comment (kevinlang) [inline lib/exqlite/connection.ex:443]: @warmwaffles I'm curious your thoughts here. I think we need the extra info for help in decoding the results, but can't find where that lives, and making this change didn't result in any tests failing.

We may need to do this normalization only within the Ecto layer, instead.

Comment (warmwaffles) [reply lib/exqlite/connection.ex:443]: Yea we can make that change in the nif. It's fairly easy.

**Comment by warmwaffles** [comment]
Let me add that to this PR real quick

**Comment by warmwaffles** [comment]
@kevinlang made a quick commit where `columns/2` just returns the column names in a list.

File: integration_test/exqlite/test_helper.exs [hunk 2]
```diff
@@ -41,9 +56,15 @@ _ = Ecto.Adapters.Exqlite.storage_down(TestRepo.config())
...
 
 :ok = Ecto.Migrator.up(TestRepo, 0, Ecto.Integration.Migration, log: false)
 Ecto.Adapters.SQL.Sandbox.mode(TestRepo, :manual)
 Process.flag(:trap_exit, true)
 
-ExUnit.start()
+ExUnit.start(
+  exclude: [
+    # SQLite does not have an array type
+    :array_type,
```
Comment (warmwaffles) [inline integration_test/exqlite/test_helper.exs:68]: It doesn't have an array type, but I specified it to get serialized as a json array. Not sure if this is just for the postgres specific array operation stuff.

File: lib/exqlite/sqlite3.ex [hunk 3]
```diff
@@ -113,5 +113,6 @@ defmodule Exqlite.Sqlite3 do
   defp convert(%DateTime{} = val), do: DateTime.to_iso8601(val)
   defp convert(%Time{} = val), do: Time.to_iso8601(val)
   defp convert(%NaiveDateTime{} = val), do: NaiveDateTime.to_iso8601(val)
+  defp convert(nil), do: "NULL"
```
Comment (warmwaffles) [inline lib/exqlite/sqlite3.ex:116]: Oof, how did I miss this.

**Review by warmwaffles — APPROVED** [review]

File: integration_test/exqlite/test_helper.exs (hunk 2 above)
Comment (warmwaffles) [reply integration_test/exqlite/test_helper.exs:68]: Yea there is no way we can support that. Unfortunate.

**Pull request merged**